resources/js/common/utils/l10n/loadDayjsLocale.ts

```diff
@@ -6,30 +6,27 @@ import dayjs from 'dayjs';
  * the client-side JS bundle size down.
  */
 export async function loadDayjsLocale(userLocale: string) {
-  if (userLocale === 'pt_BR') {
-    try {
-      await import('dayjs/locale/pt-br.js');
-      dayjs.locale('pt-br');
-    } catch (err) {
-      console.warn('Unable to load Day.js locale for pt_BR.', err);
-    }
-  }
+  try {
+    switch (userLocale) {
+      case 'es_ES':
+        await import('dayjs/locale/es.js');
+        dayjs.locale('es');
+        break;
 
-  if (userLocale === 'pl_PL') {
-    try {
-      await import('dayjs/locale/pl.js');
-      dayjs.locale('pl');
-    } catch (err) {
-      console.warn('Unable to load Day.js locale for pl_PL.', err);
-    }
-  }
+      case 'pl_PL':
+        await import('dayjs/locale/pl.js');
+        dayjs.locale('pl');
+        break;
+
+      case 'pt_BR':
+        await import('dayjs/locale/pt-br.js');
+        dayjs.locale('pt-br');
+        break;
 
-  if (userLocale === 'es_ES') {
-    try {
-      await import('dayjs/locale/es.js');
-      dayjs.locale('es');
-    } catch (err) {
-      console.warn('Unable to load Day.js locale for es_ES.', err);
+      default:
+        console.warn(`Locale ${userLocale} is not supported.`);
```

Comment:
What happens when a locale is not supported besides logging this message? Does it default to `en_US`? Also, shouldn't `en_US` have a case to prevent logging this message?

Reply:
Yes - the UI always falls back to en_US in that circumstance:
![Screenshot 2024-11-13 at 6 30 01 PM](https://github.com/user-attachments/assets/671d63d8-bf5c-46ba-9708-06dd051750f8)

However, there is indeed a bug where if the locale is en_US, the UI logs this locale is unsupported. In latest, I've fixed this bug and added the test case you've recommended to verify the fix.